src/aggregator/client/queue.go

```diff
@@ -225,52 +249,15 @@ func (q *queue) writeAndReset() {
 	} else {
 		q.metrics.connWriteSuccesses.Inc(1)
 	}
-	q.buf = q.buf[:0]
-}
 
-func (q *queue) drain() {
-	defer q.conn.Close()
-	timer := time.NewTimer(q.batchFlushDeadline)
-	lastDrain := time.Now()
-	write := func() {
-		q.writeAndReset()
-		lastDrain = time.Now()
-	}
-
-	for {
-		select {
-		case qitem := <-q.bufCh:
-			drained := false
-			msg := qitem.Bytes()
-			if len(q.buf)+len(msg) > q.maxBatchSize {
-				write()
-				drained = true
-			}
-			q.buf = append(q.buf, msg...)
-			qitem.Close()
-
-			if drained || (len(q.buf) < q.maxBatchSize &&
-				time.Since(lastDrain) < q.batchFlushDeadline) {
-				continue
-			}
-
-			write()
-		case ts := <-timer.C:
-			delta := ts.Sub(lastDrain)
-			if delta < q.batchFlushDeadline {
-				timer.Reset(q.batchFlushDeadline - delta)
-				continue
-			}
-			write()
-			timer.Reset(q.batchFlushDeadline)
-		case <-q.doneCh:
-			return
-		}
+	if cap(*buf) < _queueMaxWriteBufSize {
```

Comment:
Don't we want to reset the buffer upon successful write regardless of any checks?

Reply:
Not immediately clear what the goal is here - we don't want to pool slices we've expanded past the max buffer size? Is this just to minimize unexpected memory growth and keep things bounded? Can you add a quick comment?